### PR TITLE
Specify watch and workdir relationship in spin.toml

### PR DESCRIPTION
**Pull request by mikkelhegn**
Clarification on how workdir impacts the watch specification in spin.toml.

### CODE REVIEW

**Comment by mikkelhegn** [comment]
Not sure why the GoDadday link produces a 403. I cannot reproduce that locally.

**Comment by mikkelhegn** [comment]
For build failure #781

File: content/spin/running-apps.md [hunk 1]
```diff
@@ -104,6 +104,8 @@ command = "cargo build --target wasm32-wasi --release"
 watch = ["src/**/*.rs", "Cargo.toml"]
 ```
 
+Note that references to files in the `component.build.watch` section use the optional `component.build.workdir` section as the base directory for references. If not specified, it uses the directory of the `spin.toml` file.
```
Comment (itowlson) [inline content/spin/running-apps.md:107]: I got a bit lost in the references to references - I reckon we could simplify the wording e.g.

```suggestion
If the `build` section specifies a `workdir`, then `watch` patterns are relative to that directory. Otherwise, `watch` patterns are relative to the directory containing the `spin.toml` file.
```

**Comment by mikkelhegn** [comment]
I need to follow up on the build failures, and will take you comment in to considerations in a follow-up merge @itowlson. Good suggestion.

**Comment by mikkelhegn** [comment]
See this got incorporated here: https://github.com/fermyon/developer/pull/783/files 👍🏻 @tpmccallum
